### PR TITLE
API: Hide exceptions from the main namespace

### PR DESCRIPTION
**Pull request by seberg**
I wasn't sure if we should already start deprecating the exceptions so opted to follow up with only hiding them from `__dir__()` but still having them in `__all__` and available.

This also changes their module to `numpy.exceptions`, which matters because that is how they will be pickled (it would not be possible to unpickle such an exception in an older NumPy version).

Due to pickling, we could put off changing the module.

### CODE REVIEW

**Comment by rgommers** [comment]
Hiding but not yet deprecating seems right to me.

**Comment by seberg** [comment]
Ah, I should follow up with a scan for changing the current uses at least in documentation, and also internal uses.  But maybe its actually nice to keep it as distinct PRs.

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -886,9 +886,9 @@ cdef class Generator:
             greater than or equal to low.  The default value is 0.
         high : float or array_like of floats
             Upper boundary of the output interval.  All values generated will be
-            less than high.  The high limit may be included in the returned array of 
-            floats due to floating-point rounding in the equation 
-            ``low + (high-low) * random_sample()``.  high - low must be 
+            less than high.  The high limit may be included in the returned array of
+            floats due to floating-point rounding in the equation
+            ``low + (high-low) * random_sample()``.  high - low must be
```
Comment (mattip) [inline numpy/random/_generator.pyx:891]: did you mean to make all these whitespace cleanup changes in `_generator.pyx`?

Comment (seberg) [reply numpy/random/_generator.pyx:891]: Ack, thats a lot of whitespace fixes for one line of changed exception...

**Comment by mattip** [comment]
Thanks @seberg

**Pull request merged**